pkg/controller/migstorage/validation.go

```diff
@@ -21,12 +30,103 @@ const (
 
 // Messages
 const (
-	ReadyMessage = "The storage is ready."
+	ReadyMessage                        = "The storage is ready."
+	InvalidBackupStorageProviderMessage = "The `backupStorageProvider` must be: (aws|gcp|azure)."
+	InvalidSettingsMessage              = "The `backupStorageProvider` settings [%s] not valid."
 )
 
 // Validate the storage resource.
 // Returns error and the total error conditions set.
-func (r ReconcileMigStorage) validate(plan *migapi.MigStorage) (error, int) {
+func (r ReconcileMigStorage) validate(storage *migapi.MigStorage) (error, int) {
 	totalSet := 0
+
+	// Backup location provider.
+	err, nSet := r.validateStorageProvider(storage)
+	if err != nil {
+		return err, 0
+	}
+	totalSet += nSet
+
+	// Apply changes.
+	err = r.Update(context.TODO(), storage)
+	if err != nil {
+		return err, 0
+	}
+
 	return nil, totalSet
 }
+
+func (r ReconcileMigStorage) validateStorageProvider(storage *migapi.MigStorage) (error, int) {
+	provider := storage.Spec.BackupStorageProvider
+	var err error
+	nSet := 0
+
+	switch provider {
```

Comment:
+1, I was wondering if you were narrowing down checks to only look at the supplied BackupStorageProvider relevant settings.